### PR TITLE
Updated blog for post mid-term evaluation period

### PR DESCRIPTION
**Pull request by waridrox**
Here's the updated blog post, also incorporating mid-term evaluation period. Originally first published at [medium](https://medium.com/@waridrox/gsoc-2022-final-project-report-c206883936fa).

**Project:** CERNBox
**Title:** Etherpad plugin as a ScienceMeshDocs editor
**Author:** Mohammad Warid

### CODE REVIEW

File: _gsocblogs/2022/blog_EtherpadPlugin_MohammadWarid.md [hunk 1]
```diff
@@ -46,7 +46,33 @@ This implementation served well for the project which meant that the metadata ca
 
 Currently, I am investigating the WOPI server functionality for receiving the metadata and handling it, after which, I will move towards testing the complete workflow.
 
+# Post Midterm Period
+
+## Support for multi-user collaboration
+The main goal for the project was an in-app workflow, targeted to onboard new users over the etherpad each possessing a unique ID. This was achieved by fetching query parameters from a custom `/setEFSSMetadata` endpoint, thereby passing successful or error based responses to the wopiserver, which then allocated a new url to serve the etherpad user.
```
Comment (glpatcern) [inline _gsocblogs/2022/blog_EtherpadPlugin_MohammadWarid.md:52]: On top of what Andrei asked, please change `in-app` into `open file in app`, and then I suggest `targeted to enable users of a cloud storage opening and working with their files into Etherpad`.

File: _gsocblogs/2022/blog_EtherpadPlugin_MohammadWarid.md [hunk 2]
```diff
@@ -46,7 +46,33 @@ This implementation served well for the project which meant that the metadata ca
...
 Currently, I am investigating the WOPI server functionality for receiving the metadata and handling it, after which, I will move towards testing the complete workflow.
 
+# Post Midterm Period
+
+## Support for multi-user collaboration
+The main goal for the project was an in-app workflow, targeted to onboard new users over the etherpad each possessing a unique ID. This was achieved by fetching query parameters from a custom `/setEFSSMetadata` endpoint, thereby passing successful or error based responses to the wopiserver, which then allocated a new url to serve the etherpad user. 
+The endpoint was protected against validation checks for bogus inputs like invalid `padID` and `API` key.
+
+## Debouncing padUpdate event
+The `padUpdate` event in etherpad is supposed to fire on every new change to the etherpad. This is more noticeable when dealing with every keystroke event. To lower the amount of `pending 202` requests at the wopiserver’s end, the `padUpdate` event was debounced by a factor of 3000 milliseconds. After this implementation, only the latest updated change was taken into account.
```
Comment (glpatcern) [inline _gsocblogs/2022/blog_EtherpadPlugin_MohammadWarid.md:56]: Let's simplify `To lower the amount of pending requests` into `To lower the amount of requests` - As you know the wopiserver is performing its own "debouncing" and that's the reason it responds with 202.

**Review by glpatcern — CHANGES_REQUESTED** [review]
Thanks @waridrox, a few more suggestions follow

**Comment by waridrox** [comment]
Done!

**Comment by glpatcern** [comment]
For me it's good to go, @agheata you can merge if you wish!

**Review by agheata — APPROVED** [review]

**Pull request merged**